src/router/assignments.router.ts

```diff
@@ -0,0 +1,54 @@
+// Libraries
+import express from 'express'
+
+// Middleware
+import validator from '../middleware/validator/assignments.validator'
+import serializer from '../middleware/serializer/assignments.serializer'
+import { idAsInt } from '../middleware/validator/generic.validator'
+
+// Controller
+import AssignmentsController from '../controller/assignments.controller'
+
+const Router = express.Router()
+
+/**
+ * @swagger
+ * /assignments:
+ *   get:
+ *     summary: Retrieve a list of assignments
+ */
+Router.get('/', AssignmentsController.get, serializer)
+
+/**
+ * @swagger
+ * /assignments/{id}:
+ *   get:
+ *     summary: Retrieve a single assignment
+ */
+Router.get('/:id', idAsInt, AssignmentsController.detail, serializer)
+
+/**
+ * @swagger
+ * /assignments:
+ *   post:
+ *     summary: Create an assignment
+ */
+Router.post('/', validator, AssignmentsController.post, serializer)
```

Comment:
This will need some more validation on the user to be sure they're allowed to make new assignments for a course. But because we don't quite have that ironed out yet, I think this is fine for now.